### PR TITLE
296 scheduler fails unable to find assembly contigs

### PR DESCRIPTION
**Pull request by mbthornton-lbl**
This PR provides fixes to the scheduler responsible for incorrect behaviors around finding new jobs and creating a new entry in the `jobs` collection
#296 

Main cause of failure was a lack of matching workflow_execution db records to their workflow_config by `type` in `workflow_process.py`
this was not previously necessary, since each workflow_execution type was in its own collextion.

Other updates include
- More detailed unit tests and fixtures replicating the issue in #296 
- Refactoring for clarity
- update dependencies to nmdc-schema 11,1,0

Test Results running scheduler in local environment with local copy of NMDC DB:
```
INFO:root:Initializing Scheduler
INFO:root:Found 1 new jobs for nmdc:wfmgan-11-6x59p192.2
INFO:root:JOB RECORD: nmdc:56493f34-a9f7-11ef-b8be-d61879508757
``` 

Job Record created in local `jobs` table:
```
{
    "_id" : ObjectId("67426cfc21a22e114d84b5c3"),
    "workflow" : {
        "id" : "MAGs: v1.3.12"
    },
    "id" : "nmdc:56493f34-a9f7-11ef-b8be-d61879508757",
    "created_at" : ISODate("2024-11-23T16:02:04.000+0000"),
    "config" : {
        "git_repo" : "https://github.com/microbiomedata/metaMAGs",
        "release" : "v1.3.12",
        "wdl" : "mbin_nmdc.wdl",
        "activity_id" : "nmdc:wfrbt-11-7r0srx57.9",
        "activity_set" : "workflow_execution_set",
        "was_informed_by" : "nmdc:omprc-11-cegmwy02",
        "trigger_activity" : "nmdc:wfmgan-11-6x59p192.2",
        "iteration" : NumberInt(9),
        "input_prefix" : "nmdc_mags",
        "inputs" : {
            "proj" : "{workflow_execution_id}",
            "contig_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgan-11-6x59p192.2/nmdc_wfmgan-11-6x59p192.2_contigs.fna",
            "sam_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgas-11-jchk0x71.1/nmdc_wfmgas-11-jchk0x71.1_pairedMapped_sorted.sam.gz",
            "gff_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgan-11-6x59p192.2/nmdc_wfmgan-11-6x59p192.2_functional_annotation.gff",
            "proteins_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgan-11-6x59p192.2/nmdc_wfmgan-11-6x59p192.2_proteins.faa",
            "cog_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgan-11-6x59p192.2/nmdc_wfmgan-11-6x59p192.2_cog.gff",
            "ec_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgan-11-6x59p192.2/nmdc_wfmgan-11-6x59p192.2_ec.tsv",
            "ko_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgan-11-6x59p192.2/nmdc_wfmgan-11-6x59p192.2_ko.tsv",
            "pfam_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgan-11-6x59p192.2/nmdc_wfmgan-11-6x59p192.2_pfam.gff",
            "tigrfam_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgan-11-6x59p192.2/nmdc_wfmgan-11-6x59p192.2_tigrfam.gff",
            "crispr_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgan-11-6x59p192.2/nmdc_wfmgan-11-6x59p192.2_crt.crisprs",
            "product_names_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgan-11-6x59p192.2/nmdc_wfmgan-11-6x59p192.2_product_names.tsv",
            "gene_phylogeny_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgan-11-6x59p192.2/nmdc_wfmgan-11-6x59p192.2_gene_phylogeny.tsv",
            "lineage_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgan-11-6x59p192.2/nmdc_wfmgan-11-6x59p192.2_scaffold_lineage.tsv",
            "map_file" : "https://data.microbiomedata.org/data/nmdc:omprc-11-cegmwy02/nmdc:wfmgan-11-6x59p192.2/nmdc_wfmgan-11-6x59p192.2_contig_names_mapping.tsv"
        },

    },
    "claims" : [

    ]
}

```

### CODE REVIEW

File: nmdc_automation/workflow_automation/sched.py [hunk 1]
```diff
@@ -347,4 +346,5 @@ def main():  # pragma: no cover
 
 if __name__ == "__main__":  # pragma: no cover
     logging.basicConfig(level=logging.INFO)
-    main()
+    # site_conf and wf_file are passed in as arguments
+    main("site_configuration.toml", "workflows.yaml")
```
Comment (aclum) [inline nmdc_automation/workflow_automation/sched.py:350]: How will metatranscriptomes be handled? That has a different yaml name.

**Review by aclum — APPROVED** [review]
There doesn't appear to be a way to configure which yaml gets passed to sched.py

**Review by aclum — CHANGES_REQUESTED** [review]
There is not a way to specify what yaml sched.py uses.

File: nmdc_automation/workflow_automation/sched.py (hunk 1 above)
Comment (mbthornton-lbl) [reply nmdc_automation/workflow_automation/sched.py:350]: Fix main() argument handling

**Comment by aclum** [comment]
Based on your local results there are two issues: 
1) the minted typecode is not correct
  "activity_id" : "nmdc:wfrbt-11-7r0srx57.9", <- this should use a mag type code 'wfmag'
  2) "inputs" : {
            "proj"  : "{workflow_execution_id}", <- this variable isn't be resolved to an identifier, it should match the activity_id

**Comment by mbthornton-lbl** [comment]
[nmdc-omprc-13-01jx8727_mags_job.json](https://github.com/user-attachments/files/17907792/nmdc-omprc-13-01jx8727_mags_job.json)

**Review by aclum — APPROVED** [review]

**Pull request merged**